src/object/cli_ec.c

```diff
@@ -1662,17 +1673,33 @@ obj_ec_req_reasb(daos_iod_t *iods, d_sg_list_t *sgls, daos_obj_id_t oid,
 			goto out;
 		}
 
-		rc = obj_ec_recx_reasb(&iods[i], sgls ? &sgls[i] : NULL, oca,
-				       reasb_req, i, update, &tgt_nr);
+		rc = obj_ec_recx_reasb(&iods[i], sgls ? &sgls[i] : NULL, oca, reasb_req, i,
+				       update, &data_tgt_idx, &single_data_tgt);
 		if (rc) {
 			D_ERROR(DF_OID" obj_ec_recx_reasb failed %d.\n",
 				DP_OID(oid), rc);
 			goto out;
 		}
-		valid_tgt_nr = max(valid_tgt_nr, tgt_nr);
 	}
 
-	reasb_req->orr_single_tgt = valid_tgt_nr == 1;
+	if (single_data_tgt) {
+		struct obj_io_desc	*oiod;
+		struct obj_shard_iod	*siod;
+
+		/* if with single data target, zero the offset as each target start from same sgl
+		 * (user original input sgl).
+		 */
+		for (i = 0; i < iod_nr; i++) {
+			oiod = &reasb_req->orr_oiods[i];
+			if (oiod->oiod_siods == NULL)
+				continue;
+			for (j = 0; j < oiod->oiod_nr; j++) {
+				siod = &oiod->oiod_siods[j];
+				siod->siod_off = 0;
+			}
+		}
+	}
+	reasb_req->orr_single_tgt = single_data_tgt;
```

Comment:
sorry, not sure I follow here.   why do you need zero the offset?

Reply:
for EC partial update case, if update to one data target and replicate to all parity tgts, as we use the original user sgl to create bulk handle, so on parity shard server-side different targets should bulk transferring from 0 offset.
in current master code, we use re-assembled sgl/bulk so need the calculated offset.

Reply:
I see, thanks. hmm, though I thought siod_off is initialized by 0.